include/rcutils/types/string_array.h

```diff
@@ -31,6 +31,7 @@ typedef struct RCUTILS_PUBLIC_TYPE rcutils_string_array_t
 {
   size_t size;
   char ** data;
+  rcutils_allocator_t allocator;
```

Comment:
is that a rule that the types should hold the allocator object?
If so, ignore the comments on the `rmw` PR.